### PR TITLE
DD-272-update-dependencies

### PR DESCRIPTION
**Pull request by KLV96**
Drop Node 14, 16 and 18 and allowing only 20.x || 22.x and updating the following dependencies: 
debug ^4.3.6 -> ^4.3.7
eslint ^8.57.0 -> ^9.12.0 and replacing .eslintrc with eslint.config.js 
hmpo-logger ^7.0.2 -> ***
mocha ^10.7.0 -> ^10.7.3
nyc ^17.0.0 -> ^17.1.0
proxyquire ^2.0.0 -> ^2.1.3
sinon ^18.0.0 -> ^19.0.2
Adding Husky ^9.1.6 then adding ./husky/pre-push
globals ^15.9.0
http-proxy-agent ^5.0.0 -> ^7.0.2
https-proxy-agent ^5.0.1 -> ^7.0.5

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,34 +19,37 @@
...
   "bugs": {
     "url": "https://github.com/HMPO/hmpo-model/issues"
   },
   "homepage": "https://github.com/HMPO/hmpo-model#readme",
   "dependencies": {
-    "debug": "^4.3.6",
+    "debug": "^4.3.7",
     "got": "<12",
     "http-proxy-agent": "^5.0.0",
     "https-proxy-agent": "^5.0.1",
```
Comment (SamChatfield) [inline package.json:32]: There's 7.x versions available for both of these now which we should upgrade to if we can

File: package.json [hunk 2]
```diff
@@ -18,34 +19,37 @@
...
   "devDependencies": {
     "chai": "^4.5.0",
-    "eslint": "^8.57.0",
-    "hmpo-logger": "^7.0.2",
-    "mocha": "^10.7.0",
-    "nyc": "^17.0.0",
-    "proxyquire": "^2.0.0",
-    "sinon": "^18.0.0",
-    "sinon-chai": "^3.7.0"
+    "eslint": "^9.11.1",
```
Comment (SamChatfield) [inline package.json:37]: Version 9.12.0 now available that we can go to

**Review by SamChatfield — CHANGES_REQUESTED** [review]

File: lib/remote-model.js [hunk 3]
```diff
@@ -150,11 +150,11 @@ class RemoteModel extends LocalModel {
         const isHttps = (new URL(url).protocol === 'https:');
 
         if (isHttps) {
-            const HttpsProxyAgent = require('https-proxy-agent');
+            const { HttpsProxyAgent } = require('https-proxy-agent');
             const agent = new HttpsProxyAgent(options);
             return { https: agent };
         } else {
-            const HttpProxyAgent = require('http-proxy-agent');
+            const { HttpProxyAgent } = require('http-proxy-agent');
```
Comment (SamChatfield) [inline lib/remote-model.js:157]: The proxy agent constructors here don't seem to match the changed API introduced in 6.0.0 so calls through a proxy are most likely broken. Tests might not be working properly if they're passing on this.

https://github.com/TooTallNate/proxy-agents/blob/main/packages/https-proxy-agent/CHANGELOG.md#600
https://github.com/TooTallNate/proxy-agents/blob/main/packages/http-proxy-agent/CHANGELOG.md#600

**Review by SamChatfield — CHANGES_REQUESTED** [review]

File: .eslintrc
Comment (SamChatfield) [inline .eslintrc:1]: `test/.eslintrc` also needs removing

**Review by SamChatfield — CHANGES_REQUESTED** [review]

**Review by cmcgee480 — APPROVED** [review]

**Review by SamChatfield — APPROVED** [review]
LGTM

**Review by imran-gov — APPROVED** [review]

**Pull request merged**